evalml/pipelines/components/transformers/preprocessing/decomposer.py

```diff
@@ -205,7 +205,7 @@ def set_period(self, X: pd.DataFrame, y: pd.Series):
 
         """
         self.period = self.determine_periodicity(X, y)
-        self.parameters.update({"period": self.period})
+        self.update_parameters({"period": self.period})
```

Comment:
can you add a test verifying the period updates?

Reply:
There's a line in `test_decomposer.py::test_decomposer_set_period` that's currently commented out testing this, you should be able to uncomment that!

Reply:
I uncommented that line, I think that should be sufficient? @jeremyliweishih I can also break it out into its own separate test if need be.

Reply:
@christopherbunn thats great!